pkg/updater/repository/repository.go

```diff
@@ -0,0 +1,324 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package repository contains the packaging logic of the updater.
+package repository
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+const (
+	previousVersionLink   = "previous"
+	stableVersionLink     = "stable"
+	experimentVersionLink = "experiment"
+)
+
+// Repository contains the stable and experimental package of a single artifact managed by the updater.
+//
+// On disk the repository is structured as follows:
+// .
+// ├── 7.50.0
+// ├── 7.51.0
+// ├── stable -> 7.50.0 (symlink)
+// └── experiment -> 7.51.0 (symlink)
+//
+// We voluntarily do not load the state of the repository in memory to avoid any bugs where
+// what's on disk and what's in memory are not in sync.
+// All the functions of the repository are "atomic" and ensure no invalid state can be reached
+// even if an error happens during their execution.
+// It is possible to end up with garbage left on disk if an error happens during some operations. This
+// is cleaned up during the next operation.
+type Repository struct {
+	RootPath string
+}
+
+// State is the state of the repository.
+type State struct {
+	Stable     string
+	Experiment string
+}
+
+// HasStable returns true if the repository has a stable package.
+func (s *State) HasStable() bool {
+	return s.Stable != ""
+}
+
+// HasExperiment returns true if the repository has an experiment package.
+func (s *State) HasExperiment() bool {
+	return s.Experiment != ""
+}
+
+// GetStatus returns the status of the repository.
+func (r *Repository) GetStatus() (*State, error) {
```

Comment:
```suggestion
func (r *Repository) GetState() (*State, error) {
```

Since we changed the name of the struct, I think we should update the function name and the places where we call it. Also, the variable names as well in the case of `NewUpdater`

Reply:
Ah right missed, this one, fixed!